src/stdio/src/helpers.rs

```diff
@@ -0,0 +1,159 @@
+use super::{internal, BUFSIZ, FILE, UNGET};
+use compiler_builtins::mem::memset;
+use stdlib::calloc;
+use core::{mem, ptr};
+use core::sync::atomic::AtomicBool;
+use platform::types::*;
+use super::constants::*;
+use fcntl::*;
+use platform;
+use errno;
+
+/// Parse mode flags as a string and output a mode flags integer
+pub unsafe fn parse_mode_flags(mode_str: *const c_char) -> i32 {
+    use string::strchr;
+    let mut flags = if !strchr(mode_str, b'+' as i32).is_null() {
+        O_RDWR
```

Comment:
From the `fopen` man page

> a+    Open for reading and appending (writing at end of file).

Using `O_RDWR` here means that `a+` for us means "Open for reading, writing, and appending". Is that intentional?

Reply:
the `+` means "update", which means both read and write are available. `O_RDWR` is the correct mode in this case